.Dislocation Motion.tmp.nlogo

```diff
@@ -275,104 +283,113 @@ to calculate-fl-positions
   ]
 end
 
-to identify-force-atoms
+to identify-force-atoms ; (find the atoms closest to the force line that will be the ones receiving the external force)
   (ifelse force-mode = "Shear" [
-    ask atoms [ set ex-force-applied?  0 ]
-    ask min-n-of num-forced-atoms atoms with [ ycor >= median-ycor ] [ distancexy upper-left-fl ycor] [
-      set ex-force-applied?  1
+    ask atoms [ set ex-force-applied?  False ]
+    let forced-atoms atoms with [ ycor >= median-ycor and (distancexy upper-left-fl ycor) <= 1]
+    set num-forced-atoms count forced-atoms
+    ask forced-atoms [
+      set ex-force-applied?  True
     ]
     ]
     force-mode = "Compression" [
-      ask atoms [ set ex-force-applied?  0 ]
-      ask min-n-of num-forced-atoms atoms [ distancexy left-fl ycor] [
-        set ex-force-applied?  1
+      ask atoms [ set ex-force-applied?  False ]
+      let forced-atoms atoms with [ (distancexy left-fl ycor) <= 1]
+      set num-forced-atoms count forced-atoms
+      ask forced-atoms [
+        set ex-force-applied?  True
     ]
   ]) ; for tension, the same atoms in the left shoulder of the sample always receive the force
 end
 
 
 to check-eq-adj-force ; (check equilibrium, adjust force)
-  if f-app != f-app-prev [ set f-app-auto f-app ] ; updates f-app-auto if the user manually changed f-app
-  if precision prev-length 4 = precision (right-fl - left-fl) 4 [ set f-app-auto f-app-auto + .1 ] ; increments f-app-auto if the sample has reached an equilibrium
-  set prev-length (right-fl - left-fl)
-  set f-app-prev f-app
+  if precision prev-length 6 >= precision (right-edge - left-fl) 6 [ set f-app precision (f-app + .01) 3 ]
+  ; increments f-app-auto if the sample has reached an equilibrium
+  set prev-length (right-edge - left-fl)
 end
 
 to update-force-and-velocity-and-links
   let new-fx 0
   let new-fy 0
-  let total-force 0
-  let in-radius-atoms (other atoms in-radius cutoff-dist)
+  let total-potential 0
```

Comment:
to not confuse students, lets call this total-PE
see: https://webhome.phy.duke.edu/~schol/phy152/faqs/faq4/node8.html

Also, let's have atoms store total-PE as a turtle-variable so it is easy to graph if we want to.

Reply:
I went for total-potential-energy in lieu of total-potential, and then total-PE for the turtle-variable